source/core/src/main/com/csse3200/game/components/results/ResultsDisplay.java

```diff
@@ -0,0 +1,240 @@
+package com.csse3200.game.components.results;
+
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.graphics.Texture;
+import com.badlogic.gdx.graphics.g2d.SpriteBatch;
+import com.badlogic.gdx.scenes.scene2d.Actor;
+import com.badlogic.gdx.scenes.scene2d.InputListener;
+import com.badlogic.gdx.scenes.scene2d.ui.Image;
+import com.badlogic.gdx.scenes.scene2d.ui.Table;
+import com.badlogic.gdx.scenes.scene2d.ui.TextButton;
+import com.badlogic.gdx.scenes.scene2d.utils.ChangeListener;
+import com.badlogic.gdx.utils.Align;
+import com.csse3200.game.GdxGame;
+import com.csse3200.game.GdxGame.ScreenType;
+import com.csse3200.game.ui.UIComponent;
+import com.rafaskoberg.gdx.typinglabel.TypingAdapter;
+import com.rafaskoberg.gdx.typinglabel.TypingLabel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class ResultsDisplay extends UIComponent {
+    private static final Logger logger = LoggerFactory.getLogger(ResultsDisplay.class);
+    private static final float Z_INDEX = 2f;
+    /**
+     * The Image that forms the background of the page
+     */
+    private Image background;
+    private Table table;
+    private final GdxGame game;
+
+    private boolean typingAnimationComplete;
+    private boolean hasWon;
+
+    private TextButton returnButton;
+
+    private boolean animationFinished = false;
+
+    /**
+     * The time in seconds that it takes for the narration animation to reach the trigger point,
+     * at which the planet should enter the frame.
+     */
+    private static float textAnimationDuration = 12;
+
+    /**
+     * The speed in pixels/frame that the background and the planet should move at.
+     */
+    private float spaceSpeed = 0.33f;
+
+    /**
+     * The distance away from the top the text that the planet should stop in pixels.
+     */
+    private float planetToTextPadding = 100;
+
+    /**
+     * The Image that contains the animated planet.
+     */
+    private Image planet;
+
+    /**
+     * The table that forms the basis for the layout of the textual elements on the screen.
+     */
+    private Table rootTable;
+
+    /**
+     * The TypingLabel that contains the story that is displayed on the screen.
+     * TypingLabel is a superclass of Label, which allows Label text to be animated.
+     */
+    private TypingLabel storyLabel;
+
+    public static String losingMessage;
+
+    public ResultsDisplay(GdxGame game) {
+        super();
+        this.game = game;
+    }
+
+    @Override
+    public void create() {
+        super.create();
+        addActors();
+    }
+
+    private void addActors() {
+        this.hasWon = true;
+        background = new Image(new Texture(Gdx.files.internal("images/intro_background_v2.png")));
```

Comment:
Possibly use the `ResourceService` to make it easier to load textures, especially when they are used more than once.